vuejs/src/org/jetbrains/vuejs/intentions/expandVModel/VueExpandVModelIntention.kt

```diff
@@ -0,0 +1,68 @@
+package org.jetbrains.vuejs.intentions.expandVModel
+
+import com.intellij.lang.javascript.intentions.JavaScriptIntention
+import com.intellij.openapi.application.WriteAction
+import com.intellij.openapi.command.CommandProcessor
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.xml.XmlAttribute
+import com.intellij.psi.xml.XmlElementType
+import org.jetbrains.vuejs.codeInsight.attributes.VueAttributeNameParser
+import org.jetbrains.vuejs.codeInsight.attributes.VueAttributeNameParser.*
+import org.jetbrains.vuejs.codeInsight.attributes.VueAttributeNameParser.VueAttributeKind.DIRECTIVE
+import org.jetbrains.vuejs.codeInsight.tags.VueElementDescriptor
+import org.jetbrains.vuejs.index.isVueContext
+
+class VueExpandVModelIntention : JavaScriptIntention() {
+    override fun getFamilyName(): String = "Expand v-model"
+    override fun getText(): String = this.familyName
+    private val validModifiers = setOf("lazy", "number", "trim")
+
+    override fun isAvailable(project: Project, editor: Editor?, element: PsiElement): Boolean {
+        return element.node.elementType == XmlElementType.XML_NAME &&
+                element.parent?.node?.elementType == XmlElementType.XML_ATTRIBUTE &&
+                element.parent.isValid &&
+                (element.parent as XmlAttribute).parent.descriptor is VueElementDescriptor &&
+                isValidVModel(element.parent as XmlAttribute) &&
+                isVueContext(element)
+    }
+
+    private fun isValidVModel(attribute: XmlAttribute): Boolean {
+        val info = VueAttributeNameParser.parse((attribute.name), attribute.parent)
+        return info.kind === DIRECTIVE &&
```

Comment:
Minor: if you do the cast below, no need to check the kind.

Reply:
My reasoning was that "kind" check is faster than cast, so better check that first.. but I can drop that if you think so..

Reply:
That's an interesting question :) I think that if `info` is already loaded into local variable (i.e. registry) the `instanceof` check will be faster than accessing that variables field and comparing it to another variables field.